pkg/agent/utils.go

```diff
@@ -263,6 +263,17 @@ func IsKubernetesVersionGe(actualVersion, version string) bool {
 	return v1.GE(v2)
 }
 
+// IsAzureLinuxCgroupv2 returns true if the VHD distro is Azure Linux and the Kubernetes
+// version is greater than or equal to 1.28.0.
+func IsAzureLinuxCgroupv2(config *datamodel.NodeBootstrappingConfiguration) bool {
```

Comment:
could we theoretically just do something like `osSku == datamodel.OSSKUAzureLinux`? since that'll be the only SKU with cgroupsv2?

this func could theoretically return true if we're on a Mariner distro in k8s 1.28 (AzureLinux is the default for this case, but could cx still specify a MarinerV2cgroupsv1 image on 1.28?)

Reply:
No just checking this condition alone will still include CgroupV1. This is because datamodel.OSSKUAzureLinux is not a one-to-one-match to Azure Linux 2.0 CgroupV2 in RP, it actually points to the overall Azure Linux 2.0. The only way to differentiate between Azure Linux 2.0 CgroupV1 and Azure Linux 2.0 CgroupV2 is to check the k8s version until the day when Azure Linux 2.0 CgroupV1 is entirely deprecated.

Reply:
discussed offline, using a static check for now

Reply:
After internal discussion, the decision is to re-implement this logic by following what was done for Ubuntu 2204 where I'll create a list that contains all Azure Linux CgorupV2 distros and use this to determine whether IsCgroupV2 is true or false.